core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -562,6 +566,38 @@ public synchronized void startInternal() throws InterruptedException, IOExceptio
       throw new IOException(errorMessage, e.getCause());
     }
     LOG.info("Started Raft Journal System in {}ms", System.currentTimeMillis() - startTime);
+    joinQuorum();
+  }
+
+  private void joinQuorum() {
+    InetSocketAddress localAddress = mConf.getLocalAddress();
+    // Send a request to join the quorum.
+    // If the server is already part of the quorum, this operation is a noop.
+    AddQuorumServerRequest request = AddQuorumServerRequest.newBuilder()
+        .setServerAddress(NetAddress.newBuilder()
+            .setHost(localAddress.getHostString())
+            .setRpcPort(localAddress.getPort()))
+        .build();
+    RaftClient client = createClient();
+    client.sendReadOnlyAsync(Message.valueOf(
+        UnsafeByteOperations.unsafeWrap(
+            JournalQueryRequest
+                .newBuilder()
+                .setAddQuorumServerRequest(request)
+                .build().toByteArray()
+        ))).whenComplete((reply, t) -> {
```

Comment:
Does `whenComplete` run even on failure?

Reply:
Yes, it always runs at the end of the action. When the action fails a throwable is given in as the second parameter.